src/layout.sh

```diff
@@ -18,15 +18,24 @@ kill_layout() {
   kill $old_pid 2> /dev/null || true;
 }
 
+# Kill old process
+kill_process() {
+  local old_pid=$1;
+  kill $old_pid 2> /dev/null || true;
+}
+
 remove_listener() {
   desktop=$1;
   [[ -z "$desktop" ]] && desktop=$(get_focused_desktop);
 
-  kill_layout "$desktop";
+  local old_pid="$(get_desktop_options "$desktop" | valueof pid)";
```

Comment:
I don't get the point of this change. It was killing the process before resetting the state, now it does that after resetting the state but what does that accomplish?

Reply:
that is a change from the original PR, not sure what the goal was here to be honest :thinking: 

and, with the changes you originally requested in #36, `kill_layout "$desktop"` and 
```bash
old_pid="$(get_desktop_options "$desktop" | valueof pid)"
kill_process "$old_pid"
```
should do exactly the same thing :+1:

Reply:
> change the way kill_layout gets called... desktop options in remove_listener have to be cleared before calling kill_layout,
> otherwise if remove_listener is called by the subscriber process on desktop_remove event, it will kill itself before the options are cleared.
>
that is the original comment on the PR :ok_hand: